src/d/actor/d_a_obj_tntrap.cpp

```diff
@@ -4,172 +4,643 @@
+#include "d/actor/d_a_ship.h"
+#include "d/d_com_inf_game.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "m_Do/m_Do_mtx.h"
+
+
+const char l_arcname[] = "TnTrap";
+
+static const dCcD_SrcTri l_tri_src = {
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0x0800,
+        /* SrcObjAt  Atp     */ 1,
+        /* SrcObjAt  SPrm    */ 5,
+        /* SrcObjTg  Type    */ 0,
+        /* SrcObjTg  SPrm    */ 0,
+        /* SrcObjCo  SPrm    */ 0,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0x00,
+        /* SrcGObjAt Spl     */ 0xB,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ 0,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    {
+        /* a */ 0.0f, 0.0f, 0.0f,
+        /* b */ 0.0f, 0.0f, 0.0f,
+        /* c */ 0.0f, 0.0f, 0.0f,
+    },
+};
+
+static const Vec l_tri_vtx[6] = {
+300.0, -70.0, 0.0, 
+-300.0, -70.0, 0.0, 
+-300.0, 55.0, 0.0, 
+300.0, 55.0, 0.0, 
+300.0, 180.0, 0.0, 
+-300.0, 180.0, 0.0,
+
+};
+
+
+
+static const f32 l_offset_ball[2][3] = {
+    -300,90,0,
+    300,90,0
+};
+static const f32 l_offset_thunder[3][3] = {
+    0,25,0,
+    0,85,0,
+    0,145,0
+
+};
+
 
 /* 00000078-000002AC       .text chk_appear__13daObjTnTrap_cFv */
-void daObjTnTrap_c::chk_appear() {
-    /* Nonmatching */
+BOOL daObjTnTrap_c::chk_appear() {
+
+    int o_retval = 0;
+    field_0xDC8 = param_get_swSave();
+    field_0xDCC = param_get_swSave2();
+    field_0xDD0 = param_get_arg0();
+    field_0xDD4 = param_get_mapType();
+    switch(field_0xDD4){
+    case 0:
+        if(dComIfGs_isEventBit(0x3A04) == 1){
+            if(field_0xDC8 != 0xFF && fopAcM_isSwitch(this,field_0xDC8) == 1){
+                if(dComIfGs_getTriforceNum() == 8 && field_0xDD0 == 0){
+                    if(dComIfGs_isEventBit(0x2C01) == 1){
+                        if(field_0xDCC != 0xFF && !fopAcM_isSwitch(this,field_0xDCC)){
+                            field_0x298 = 2;
+                            o_retval = 1;
+                        }
+                    }else{
+                        field_0x298 = 1;
+                        o_retval = 1;
+                    }
+                }
+            }else{
+                field_0x298 = 0;
+                o_retval = 1;
+            }
+        }
+        break;
+    case 1:
+        if(field_0xDC8 != 0xFF && !fopAcM_isSwitch(this,field_0xDC8)){
+            field_0x298 = 3;
+            o_retval = 1;
+        }
+        break;
+    case 2:
+        if(field_0xDC8 != 0xFF){
+            if(!fopAcM_isSwitch(this,field_0xDC8)){
+                field_0x298 = 5;
+                o_retval = 1;
+            }
+        }else{
+            field_0x298 = 5;
+            o_retval = 1;       
+        }
+        break;
+    default:   
+        JUT_ASSERT(0x17C,0);
```

Comment:
```suggestion
        JUT_ASSERT(380, FALSE);
```

Macros get expanded in the assertions, this was probably just `FALSE`. First argument to the `JUT_ASSERT` macro is a line number so keeping it in decimal might make more sense.